### DISCUSSION TITLE
How to insert text from TextEditor element?

### DISCUSSION THREAD

**Discussion by ThatXliner** [atom/atom#22073] [Q&A] [answered]
Hello! I am trying to hack on Atom. The GitHub package's commit editor is not a real editor component, so I can't get it from the `atom.workspace`. I have the following code:
```js
document.querySelector(".github-AtomTextEditor-container > atom-text-editor").getModel()
```
Which should return the GitHub commit editor TextEditor model. I want to know how, or even if it's possible, to [`insertText`](https://flight-manual.atom.io/api/v1.55.0/TextEditor/#instance-insertText) on that element.

If you require some clarification on my question, feel free to ask (but not StackOverflow-style hard, please)! 😊

**Comment by UziTech** [comment, marked as answer]
This question should probably be ask on the [github package repo](https://github.com/atom/github)

**Reply by ThatXliner** [reply]
The thing is, I'm just asking if its possible to do that for any text editor element, the GitHub commit message editor as an example. But sure, maybe I will.

**Reply by UziTech** [reply]
You should be able to use [`setText`](https://flight-manual.atom.io/api/v1.55.0/TextEditor/#instance-setText)

```js
document.querySelector(".github-AtomTextEditor-container > atom-text-editor").getModel().setText("test")
```

seems to work in the console.

**Reply by ThatXliner** [reply]
Wait... that works! Funny how it doesn't have an `.insertText` method. Oh well.

**Reply by UziTech** [reply]
It does have an `insertText` method for me. And it seems to work to insert text. Atom v1.56.0-beta0 Windows 10

![image](https://user-images.githubusercontent.com/97994/112350847-6b1dd680-8c97-11eb-9ecc-fdd6bd09d72b.png)

**Reply by ThatXliner** [reply]
I have Atom 1.54. Maybe It's introduced in 1.55. Who knows? ¯\\\_(ツ)_/¯
